internal/pkg/manifest/lb_web_svc.go

```diff
@@ -250,23 +257,42 @@ func (c *NetworkLoadBalancerConfiguration) IsEmpty() bool {
 func (lbws *LoadBalancedWebService) ExposedPorts() (ExposedPortsIndex, error) {
 	var exposedPorts []ExposedPort
 	workloadName := aws.StringValue(lbws.Name)
+	// port from image.port.
 	exposedPorts = append(exposedPorts, lbws.ImageConfig.exposedPorts(workloadName)...)
+	// port from sidecar[x].image.port.
 	for name, sidecar := range lbws.Sidecars {
 		out, err := sidecar.exposedPorts(name)
 		if err != nil {
 			return ExposedPortsIndex{}, err
 		}
 		exposedPorts = append(exposedPorts, out...)
 	}
+	// port from http.target_port.
 	exposedPorts = append(exposedPorts, lbws.RoutingRule.exposedPorts(exposedPorts, workloadName)...)
-	out, err := lbws.NLBConfig.exposedPorts(exposedPorts, workloadName)
-	if err != nil {
-		return ExposedPortsIndex{}, err
+
+	// port from nlb.target_port and nlb.additional_listeners[x].target_port
+	for _, listener := range lbws.NLBConfig.NLBListeners() {
+		out, err := listener.exposedPorts(exposedPorts, workloadName)
+		if err != nil {
+			return ExposedPortsIndex{}, err
+		}
+		exposedPorts = append(exposedPorts, out...)
 	}
-	exposedPorts = append(exposedPorts, out...)
 	portsForContainer, containerForPort := prepareParsedExposedPortsMap(sortExposedPorts(exposedPorts))
 	return ExposedPortsIndex{
 		PortsForContainer: portsForContainer,
 		ContainerForPort:  containerForPort,
 	}, nil
 }
+
+// NLBListeners returns main as well as additional listeners as a list of NetworkLoadBalancerListener.
+func (cfg NetworkLoadBalancerConfiguration) NLBListeners() []NetworkLoadBalancerListener {
+	var nlbListeners []NetworkLoadBalancerListener
+	if !cfg.MainListener.IsEmpty() {
+		nlbListeners = append(nlbListeners, cfg.MainListener)
+	}
+	nlbListeners = append(nlbListeners, cfg.AdditionalListeners...)
```

Comment:
is this valid:
```
nlb:
   additional_listeners:
      - ...
```

I suppose not?

Reply:
yeah it is not valid, but that is taken care by the validation already so we are safe here. But if you want to be extra safe then I can put this with MainListener as well.

Reply:
Yeah I think I should keep additional listeners with the main listener to make sure if main listener exists then only additional listeners are considered.

Reply:
👍🏼  In this case we can just return when `!cfg.MainListener.IsEmpty()`